components/pricing_modal/self_hosted_content.tsx

```diff
@@ -192,11 +199,12 @@ function SelfHostedContent(props: ContentProps) {
                             action: () => {
                                 trackEvent('self_hosted_pricing', 'click_upgrade_button');
                                 const professionalProduct = findSelfHostedProductBySku(products, SelfHostedProducts.PROFESSIONAL);
-                                if (isSelfHostedPurchaseEnabled && productsLoaded && professionalProduct) {
+                                if (isSelfHostedPurchaseEnabled && productsLoaded && professionalProduct && canUseSelfHostedSignup) {
```

Comment:
If self hosted purchase is not enabled, I think we should keep the `window.open(CloudLinks.SELF_HOSTED_SIGNUP, '_blank');` behavior. We don't want to make unrelated claims they can't reach our server. They might purposely disable the flag, for example, since it controls whether they can even *try* to load Stripe via content security policy (https://github.com/mattermost/mattermost-server/blob/master/web/handlers.go#L239-L241), and they might not want that for their workspace.

Reply:
Gotcha, I was a bit confused as to what was happening here.

I think I've got what you're looking for, I've pushed this change.